src/webui/www/private/rename_files.html

```diff
@@ -68,8 +68,9 @@
         }
 
         // Register keyboard events to modal window
+        var keyboard = undefined;
         if (!keyboard) {
```

Comment:
Why to check it if you assigned it to undefined just above?

Reply:
I can change it to `var keyboard;` but I wanted to be explicit.

Reply:
CodeQL report the variable wasn't declared before use.

Reply:
> CodeQL report the variable wasn't declared before use.

Although I am not familiar enough with JS stuff, but to me it looked as if `keyboard` was intended to preserve a previously assigned value (i.e., to reuse `Keyboard` object once created).

Reply:
Otherwise it is meaningless to check variable just declared as undefined, isn't it?

Reply:
>but to me it looked as if keyboard was intended to preserve a previously assigned value (i.e., to reuse Keyboard object once created).

But it wasn't declared anywhere else other than this place.

>Otherwise it is meaningless to check variable just declared as undefined, isn't it?

Or you can really view it as improperly coded.

Reply:
>Or you can really view it as improperly coded.

I meant the original code was improperly coded. Here I just make sure the variable was declared properly.
I'm not confident to make more clean ups as it might introduce unforeseen regressions...

Reply:
@loligans 
It is the code introduced by you recently. Can you comment on it?

Reply:
I think originally it was coded to guard against a global `keyboard` variable but in reality such variable wouldn't exist (most likely due to strict mode enabled).

Reply:
> I think originally it was coded to guard against a global `keyboard` variable but in reality such variable wouldn't exist (most likely due to strict mode enabled).

I would still wait a bit for the answer of the author of this code.

Reply:
It is also interesting why the following code does not cause analyzer complaints about expression that always evaluates to false:
```c++
var keyboard = undefined;
if (!keyboard) {
```

Reply:
yes this functionality is intended, it's using `variable hoisting` to achieve the intended behavior. no we do not want to explicitly set `keyboard = undefined` i'll explain in more detail below. 

`keyboard` is not always going to be `undefined`. this is because of how the modal dialog window get loaded. The modal window is loading using the `xhr` method instead of the `iframe` method. what this means is the `rename_files` page is appended to `index.html` and it's javascript code gets executed in the same context as `index.html`. the problem is any variables defined in the `<script>` tag are preserved after closing the modal. reloading the `rename_files` modal will cause the javascript code to excecute again and these variables to be redefined, but `let` and `const` variables are incapable of being redefined. that's why `var` is used and why we check `if(!keyboard)`

Reply:
Thanks for the explanation.
However the code is still a bit confusing from the fact that `use strict;` is declared at the top. This means that `keyboard` will be defined somewhere *before* running the script in `rename_files` but `keyboard` was not declared in `rename_files`.
I propose to change it to:
```js
        // https://github.com/qbittorrent/qBittorrent/pull/18687#discussion_r1135045726
        var keyboard;
        if (!keyboard) {
```
This way the `keyboard` is declared and its value won't be overwritten. @loligans what do you say?

Reply:
> I propose to change it to:
> 
> ```js
> // https://github.com/qbittorrent/qBittorrent/pull/18687#discussion_r1135045726
> var keyboard;
> if (!keyboard) {
>     keyboard = new Keyboard({
> ```

Forgive my ignorance. Doesn't `var keyboard;` override a previously declared variable?

Reply:
>Forgive my ignorance. Doesn't var keyboard; override a previously declared variable?

https://stackoverflow.com/questions/6888506/redeclaring-a-javascript-variable

Reply:
Well, if I understood "hoisting" correctly, then the current code is correct, since the variable is still declared in the scope in which it is used.

Reply:
>since the variable is still declared in the scope in which it is used.

This is the confusing point which my [previous comment](https://github.com/qbittorrent/qBittorrent/pull/18687#discussion_r1135074002) aims to address.

Reply:
> > since the variable is still declared in the scope in which it is used.
> 
> This is the confusing point which my [previous comment](https://github.com/qbittorrent/qBittorrent/pull/18687#discussion_r1135074002) aims to address.

It still looks to me like an attempt to give a slightly less confusing look to something that is confusing by nature. But I won't go into further discussions about it. It is enough that now the "fixing the analyzer warnings" does not break the logic of the program.

Reply:
>This is the confusing point which my https://github.com/qbittorrent/qBittorrent/pull/18687#discussion_r1135074002 aims to address.

commit updated.

Reply:
>It still looks to me like an attempt to give a slightly less confusing look to something that is confusing by nature. 

Yes and in the future with the analyzer in place, new code can be inspected better or even rewritten to avoid such situations.

Reply:
yeah it's pretty confusing. i believe @glassez is correct in that you can redeclare `var keyboard`